File: backend-clinica/backend-clinica/src/main/java/com/clinica/backendclinica/service/UsuarioService.java
package com.clinica.backendclinica.service;

import com.clinica.backendclinica.model.Usuario;

import java.util.List;

public interface UsuarioService {
    List<Usuario> BuscarTodos();
    Usuario BuscarPorId(Long id);
    Usuario Crear(Usuario usuario);
    Usuario Actualizar(Long id, Usuario usuario);
    void Eliminar(Long id);
    Usuario asignarRol(Usuario usuario, String nombreRol);

}

File: frontend-clinica/src/app/core/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Usuario } from '../models/usuario.model';

@Injectable({ providedIn: 'root' })
export class UsuarioService {
  private readonly apiUrl = 'http://localhost:8081/admin/usuario';

  constructor(private http: HttpClient) {}

  listar(): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(`${this.apiUrl}/listar`);
  }

  obtener(id: number): Observable<Usuario> {
    return this.http.get<Usuario>(`${this.apiUrl}/obtenerUsuario/${id}`);
  }

  crear(usuario: Usuario): Observable<Usuario> {
    return this.http.post<Usuario>(`${this.apiUrl}/crearUsuario`, usuario);
  }

  actualizar(id: number, usuario: Usuario): Observable<Usuario> {
    return this.http.put<Usuario>(`${this.apiUrl}/actualizarUsuario/${id}`, usuario);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/eliminarUsuario/${id}`);
  }
}
